texar/modules/decoders/decoder_helpers.py

```diff
@@ -455,6 +455,25 @@ def _top_k_logits(logits: torch.Tensor, k: int) -> torch.Tensor:
         torch.full_like(logits, float('-inf')), logits)
 
 
+def _top_p_logits(logits: torch.Tensor, p: float) -> torch.Tensor:
+    r"""Adapted from
+    https://gist.github.com/thomwolf/1a5a29f6962089e871b94cbd09daf317#file-top-k
```

Comment:
Please keep the URL on the same line. It will not be checked for line length.

Reply:
Resolved in d07e236